tests/unit/mixins/indexProvider.spec.js

```diff
@@ -0,0 +1,521 @@
+      mocks: {
+        $route: {
+          path: '/documentation/bar',
+        },
+      },
+    });
+    await flushPromises();
+    expect(IndexStore.state).toEqual({
+      flatChildren,
+      references,
+      apiChanges: null,
+      includedArchiveIdentifiers,
+      errorFetching: false,
+      technologyProps,
+    });
+  });
+
+  it('returns undefined technology when index response is empty', async () => {
+    const emptyResponse = { interfaceLanguages: {} };
+    fetchData.mockResolvedValue(emptyResponse);
+    createWrapper();
+    await flushPromises();
+    expect(IndexStore.state).toEqual({
+      flatChildren: {},
+      references: {},
+      apiChanges: null,
+      includedArchiveIdentifiers: [],
+      errorFetching: false,
+      technologyProps: {},
+    });
+  });
+
+  it('counts the amount of deprecated items a groupMarker has', async () => {
+    const technologyClone = JSON.parse(JSON.stringify(extendedTechnologies));
+    technologyClone.children[1].deprecated = true;
+    technologyClone.children[2].deprecated = true;
+    technologyClone.children[1].children[1].deprecated = true;
+    fetchData.mockResolvedValue({
+      interfaceLanguages: {
+        [Language.swift.key.url]: [
+          technologyClone,
+        ],
+      },
+    });
+    createWrapper();
+    await flushPromises();
+    expect(IndexStore.state.flatChildren[Language.swift.key.url][0]).toHaveProperty('deprecatedChildrenCount', 2);
+    expect(IndexStore.state.flatChildren).toMatchSnapshot();
+  });
+
+  it('removes the `beta` flag from children, if the parent is a `beta`', async () => {
+    const technologyClone = JSON.parse(JSON.stringify(extendedTechnologies));
+    technologyClone.beta = true;
+    technologyClone.children[1].beta = true;
+    technologyClone.children[1].children[0].beta = true;
+    fetchData.mockResolvedValue({
+      interfaceLanguages: {
+        [Language.swift.key.url]: [
+          technologyClone,
+        ],
+      },
+    });
+    createWrapper();
+    await flushPromises();
+    expect(IndexStore.state.flatChildren).toMatchSnapshot();
+  });
+
+  it('removes the `beta` flag from children, if the parent is a `beta`', async () => {
+    const technologyClone = JSON.parse(JSON.stringify(extendedTechnologies));
+    technologyClone.children[1].beta = true;
+    technologyClone.children[1].children[1].beta = true;
+    // case where the direct parent is NOT `Beta`, but an ancestor is
+    technologyClone.children[1].children[2].children[0].beta = true;
+    // set an end node as beta
+    technologyClone.children[2].children[0].beta = true;
+    fetchData.mockResolvedValue({
+      interfaceLanguages: {
+        [Language.swift.key.url]: [
+          technologyClone,
+        ],
+      },
+    });
+    createWrapper();
+    await flushPromises();
+    expect(IndexStore.state.flatChildren).toMatchSnapshot();
+  });
+
+  it('flattens deeply nested children and provides them to the NavigatorCard', async () => {
```

Comment:
Maybe "to a component"? I don't believe the navigator is using this mixin (just yet).